public/script.js

```diff
@@ -143,78 +140,117 @@ async function animate(extCon, norSouth, dates, monthLoop, starting, ending) {//
+                monthEnd = monthLoop;
             }
-            for (day = dayStart; day <= dayEnd; day++) {//loop days
-                if (year >=1989 || day %2 ==1){
-                    
-                    if (stop) return;//Catch to stop animation
-                    while (pause) {
-                        await sleep (100);
-                    }
-                    var monthStr = ""+month;//create the string for month, ensuring it is 2 digits
-                    if (month < 10) {
-                        monthStr = "0"+month;
-                    }
-                    var dayStr = ""+day;//string for the day, ensuring 2 digits
-                    if (day < 10) {
-                        dayStr = "0"+day;
+            for (month = monthStart; month <= monthEnd; month++) {//Loop months
+                var dayStart = 1;//default starting day
+                var dayEnd = monthDays[month-1];//default ending day, month -1 because index starts at 0
+                if (month == starting[1] && year == starting[0]) {//if on first month and year, start at the specified date
+                    dayStart = starting[2];
+                }
+                if (month == ending[1] && year == ending[0]) {//if on the last month and year, end at the last date
+                    dayEnd = ending[2];
+                }
+                if (dates == "Monthly" || dates == "SameMonth") {//If it is looping only months or the same month, just use the first day of the month
+                    dayEnd = 1;
+                    dayStart = 1;
+                }
+                if (onlyLoad){
+                    for (day = dayStart; day <= dayEnd; day++) {//loop days
+                        if (numFrames < 31){
+                            if (year >=1989 || day %2 ==1){
+                                if (stop) return;//Catch to stop animation
+                                var monthStr = ""+month;//create the string for month, ensuring it is 2 digits
+                                if (month < 10) {
+                                    monthStr = "0"+month;
+                                }
+                                var dayStr = ""+day;//string for the day, ensuring 2 digits
+                                if (day < 10) {
+                                    dayStr = "0"+day;
+                                }
+                                var yearStr = ""+year;//String for year
+                                await displayDates.push("Date: "+yearStr+"-"+monthStr+"-"+dayStr);
+                                imageURL = "https://nsidc.org/api/mapservices/NSIDC/wms?service=WMS&version=1.1.0&request=GetMap&layers=NSIDC:g02135_"+extCon+"_raster_daily_"+norSouth+"&styles=NSIDC:g02135_"+extCon+"_raster_basemap&bbox="+locationVal+"&format=image/png&TIME="+yearStr+"-"+monthStr+"-"+dayStr;
+                                await map.addLayer(new ol.layer.Image({
+                                    source: new ol.source.ImageStatic({
```

Comment:
Consider using the openlayers WMS source. This allows you to pass parameters like `TIME`, layer name, etc. without having to manipulate the URL string.

Reply:
And here's an example of updating the `TIME` parameter on the WMS source to animate weather radar data: https://openlayers.org/en/latest/examples/wms-time.html !